### PR TITLE
[feature](array) Add array_last lambda function

### LINKED ISSUE #18266: [Feature] support array_last function, get the last element of the array
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Support arrayLast(func, arr1) High-order function, which 
Returns the last element in the arr1 array for which func(arr1[i]...) returns something other than 0.
You must pass a lambda function to it as the first argument, and it can’t be omitted.
The lambda function must have UInt8 output type.

### Use case

select  arrayLast((a)->(a > 3), [0,1,2,3,4,5,6,7,8,9,1])  as last_element

with result:
9

### Related issues

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by herry2038**
# Proposed changes

Issue Number: close #18266

## Problem summary

Add array_last lambda function.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

Realize a common lambda function array_clone_expr, which only get expr value from its fist child.
Array_last will change to array_clone_expr when analyzing.

### CODE REVIEW

**Comment by herry2038** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/analysis/LambdaFunctionCallExpr.java [hunk 1]
```diff
@@ -198,12 +202,50 @@ public void analyzeImpl(Analyzer analyzer) throws AnalysisException {
...
+            argTypes = new Type[1] ;
+            argTypes[0] = getChild(0).getType();
+
+            fn = getBuiltinFunction(arrayCloneExpr.getFunction(), argTypes, Function.CompareMode.IS_NONSTRICT_SUPERTYPE_OF);
+            if (fn == null) {
+                LOG.warn("fn element_at not exists");
+                throw new AnalysisException(getFunctionNotFoundError(collectChildReturnTypes()));
+            }
+            fn.setReturnType(getChild(0).getType());
+        } else if ( fnName.getFunction().equalsIgnoreCase("array_clone_expr") ) {
```
Comment (zhangstar333) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/LambdaFunctionCallExpr.java:242]: I think is better to rewrite array_last(lambda,array)--->element_at(array_filter,-1)

**Comment by zhangstar333** [comment]
and need add some docs to introduce this function usage

File: fe/fe-core/src/main/java/org/apache/doris/analysis/LambdaFunctionCallExpr.java [hunk 2]
```diff
@@ -44,6 +46,8 @@ public class LambdaFunctionCallExpr extends FunctionCallExpr {
```
Comment (zhangstar333) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/LambdaFunctionCallExpr.java:46]: add this set also

File: be/src/vec/exprs/lambda_function/lambda_function_factory.h [hunk 3]
```diff
@@ -28,6 +28,7 @@ class LambdaFunctionFactory;
 
 void register_function_array_map(LambdaFunctionFactory& factory);
 void register_function_array_filter(LambdaFunctionFactory& factory);
+void register_function_array_clone_expr(LambdaFunctionFactory& facotry);
```
Comment (zhangstar333) [inline be/src/vec/exprs/lambda_function/lambda_function_factory.h:31]: could remove this

File: docs/zh-CN/docs/sql-manual/sql-functions/array-functions/array_last.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+## array_last
+
+<version since="2.0">
+
+array_last
+
+</version>
+
+### description
+Returns the last element in the array for which func(arr1[i]) returns something other than 0.
```
Comment (zhangstar333) [inline docs/zh-CN/docs/sql-manual/sql-functions/array-functions/array_last.md:34]: this line maybe could remove in zh-CN docs

**Review by zhangstar333 — DISMISSED** [review]

**Comment by zhangstar333** [comment]
run buildall

File: be/src/vec/exprs/lambda_function/lambda_function_factory.h (hunk 3 above)
Comment (herry2038) [reply be/src/vec/exprs/lambda_function/lambda_function_factory.h]: done

File: docs/zh-CN/docs/sql-manual/sql-functions/array-functions/array_last.md (hunk 4 above)
Comment (herry2038) [reply docs/zh-CN/docs/sql-manual/sql-functions/array-functions/array_last.md]: done

**Comment by herry2038** [comment]
run buildall

**Comment by herry2038** [comment]
Run p0

**Comment by herry2038** [comment]
Run p1

**Review by zhangstar333 — APPROVED** [review]

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**